src/lib/components/ScrollSentinel.svelte

```diff
@@ -0,0 +1,22 @@
+<script lang="ts">
+  import { layoutContentTopHidden } from "$lib/stores/layout.store";
+  import { onMount } from "svelte";
+
+  // The ScrollSentinel component should be placed right before the content
+  // inside the scrollable container.
+  export let scrollContainer: HTMLElement;
+
+  // To observe when the top leaves the view
+  let element: HTMLElement;
+
+  onMount(() => {
+    const observer = new IntersectionObserver(
+      ([entry]) => layoutContentTopHidden.set(!entry.isIntersecting),
+      { root: scrollContainer, threshold: 0 },
+    );
+    observer.observe(element);
+    return () => observer.disconnect();
```

Comment:
dumb question: same effect as registering an on destroy?

Reply:
I expect the same effect as [onUnmount](https://svelte.dev/docs/svelte/lifecycle-hooks#onMount).

Reply:
Nice! 

To be sure I double checked the doc of Svelte v4: https://v4.svelte.dev/docs/svelte#onmount